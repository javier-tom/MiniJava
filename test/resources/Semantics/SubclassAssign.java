class Main {
    public static void main(String[] args) {
        System.out.println(new SubclassAssign().run());
    }
}

class SubclassAssign {
    public int run() {
        A a;
        B b;
        a = new A();
        b = new B();
        a = b;
        b = a;
        return 5;
    }
}

class A extends B {}

class B {}
