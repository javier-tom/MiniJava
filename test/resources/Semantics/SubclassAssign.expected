class A:
  super: B
  fields:

class B:
  super: null
  fields:

class SubclassAssign:
  super: null
  fields:
  method run:
    params: 
    returns: int
    locals:
      A a;
      B b;

